crates/core/component/dex/src/component/dex.rs

```diff
@@ -83,6 +133,57 @@ impl Component for Dex {
                 .expect("handling batch swaps is infaillible");
             metrics::histogram!(crate::component::metrics::DEX_BATCH_DURATION)
                 .record(batch_start.elapsed());
+
+            bsods.insert(trading_pair, bsod);
+        }
+
+        // F.1. Having performed all batch swaps, "claim" the base fees and tips.
+        // The VCB has already been debited through the BSOD.
+        {
+            let state_ref =
+                Arc::get_mut(state).expect("should have unique ref after finishing batch swaps");
+            for (asset_id, (base_fee, tip)) in base_fees_and_tips.iter() {
+                if *asset_id == *STAKING_TOKEN_ASSET_ID {
+                    // In this case, there was nothing to swap, so there's nothing
+                    // to claim and we just accumulate the fee we took back into the fee component.
+                    state_ref.raw_accumulate_base_fee(Fee::from_staking_token_amount(*base_fee));
+                    state_ref.raw_accumulate_tip(Fee::from_staking_token_amount(*tip));
+                    continue;
+                }
+                let pair = TradingPair::new(*asset_id, *STAKING_TOKEN_ASSET_ID);
+                let bsod = bsods
+                    .get(&pair)
+                    .expect("bsod should be present for chain-submitted swap");
+
+                let (base_input, tip_input) = if pair.asset_1() == *asset_id {
+                    ((*base_fee, 0u64.into()), (*tip, 0u64.into()))
+                } else {
+                    ((0u64.into(), *base_fee), (0u64.into(), *tip))
+                };
+
+                let base_output = bsod.pro_rata_outputs(base_input);
+                let tip_output = bsod.pro_rata_outputs(tip_input);
+                tracing::debug!(
+                    ?asset_id,
+                    ?base_input,
+                    ?tip_input,
+                    ?base_output,
+                    ?tip_output,
+                    "claiming chain-submitted swap for alt fee token"
+                );
+
+                // Obtain the base fee and tip amounts in the native token, discarding any unfilled amounts.
+                let (swapped_base, swapped_tip) = if pair.asset_1() == *asset_id {
+                    (base_output.0, tip_output.0)
+                } else {
+                    (tip_output.1, base_output.1)
```

Comment:
I don't understand why this isn't:

```rust
let (swapped_base, swapped_tip) = if pair.asset_1() == *asset_id {
    // `0: asset_id, 1: staking token`
    (base_output.1, tip_output.1)
} else {
    // `0: staking token, 1: asset_id`
    (base_output.0, tip_output.0)
}
```

Reply:
We want to record the staking token output right?

Reply:
Yeah this seems bugged